minitask2/akhmad/behaviour.py

```diff
@@ -0,0 +1,196 @@
+#!usr/bin/env python
+
+import rospy
+from tf.transformations import euler_from_quaternion
+
+from sensor_msgs.msg import LaserScan
+from geometry_msgs.msg import Twist
+from nav_msgs.msg import Odometry
+
+from math import radians, sqrt
+import random
+
+move_cmd = Twist()
+move_cmd.linear.x = 0.2
+
+left_turn_cmd = Twist()
+left_turn_cmd.linear.x = 0
+left_turn_cmd.angular.z = radians(30)
+
+right_turn_cmd = Twist()
+right_turn_cmd.linear.x = 0
+right_turn_cmd.angular.z = radians(-30)
+
+save_position = True
+last_pose_theta = None
+last_pose_x = None
+last_pose_y = None
+
+continue_right = False
+continue_left = False
+
+front = None
+right_front = None
+right_back = None
+right = None
+back = None
+left_back = None
+left_front = None
+left = None
+
+is_wall_found = False
+
+
+class Behaviour():
+    def __init__(self):
+        self.behaviour = input("Enter behaviour mode '0' for random walk '1' for right hand wall following: ")
+        rospy.init_node('behaviour')
+
+        # Subscribers
+        self.laser_sub = rospy.Subscriber('/scan', LaserScan, self.scan_callback)
+        self.odom_sub = rospy.Subscriber('/odom', Odometry, self.odom_callback)
+
+        # Publisher
+        self.cmd_vel_pub = rospy.Publisher('/cmd_vel', Twist, queue_size=3)
+
+        self.rate = rospy.Rate(10)
+        self.current_pose_theta = None
+        self.current_pose_x = None
+        self.current_pose_y = None
+        rospy.spin()
+
+    def publish_command(self, msg):
+        self.cmd_vel_pub.publish(msg)
+
+    def turn_left(self):
+        global continue_left, continue_right
+        continue_right = False
+        continue_left = True
+        self.publish_command(left_turn_cmd)
+        print("Avoiding obstacle by turning left")
+
+    def turn_right(self):
+        global continue_left, continue_right
+        continue_right = True
+        continue_left = False
+        self.publish_command(right_turn_cmd)
+        print("Avoiding obstacle by turning right")
+
+    def right_hand_wall_follow(self):
+        global is_wall_found
+
+        if not is_wall_found:
```

Comment:
I think it would be cleaner to flip this condition, or (even better) just return after this top branch, then we could get rid of a level of indentation for lines `89-103`

Reply:
I will try it out